core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -1155,7 +1193,11 @@ public boolean isCompatible(String subject,
       prevParsedSchemas.add(prevParsedSchema);
     }
 
-    return parseSchema(newSchema).isCompatible(compatibility, prevParsedSchemas);
+    ParsedSchema parsedSchema = parseSchema(newSchema);
+    boolean isCompatible = parsedSchema.isCompatible(compatibility, prevParsedSchemas);
+    // Allow schema providers to modify the schema during compatibility checks
+    newSchema.setSchema(parsedSchema.canonicalString());
```

Comment:
Since the function only checks compatibility should we avoid adding side effect to `newSchema`?

Reply:
I'm allowing for use cases where the new schema can be merged with the older schema during the compat check.  I added this JavaDoc elsewhere in the PR:
`Custom providers may choose to modify this schema during this check, to ensure that it is compatible with the specified schema.`

Reply:
Gotcha. Thanks!